src/Components/DrillPageMinimal.js

```diff
@@ -0,0 +1,159 @@
+import React, { useLayoutEffect } from 'react';
+import { ScrollView, StyleSheet, View, Text, Image, TouchableOpacity } from 'react-native';
+import { connect } from 'react-redux';
+
+import Progress from './ProgressBar2';
+import DrillAnimation from './DrillAnimation';
+
+import iconCheckMark from '../Images/ic_check.png';
+
+import theme from '../styles/theme.style';
+
+const mapStateToProps = state => {
+  return {
+    drills: state.drills,
+  };
+};
+
+export const DrillPageMinimal = props => {
+  const { route, navigation, drills } = props;
+  const { drill, training } = route.params;
+
+  const totalDrills = training.drills.length;
+  const currentDrill = training.drills.findIndex(id => id === drill.id);
+
+  if (currentDrill === -1) navigation.navigate('TrainingPage', { training });
+
+  const onProgressDotPress = idx => {
+    const drill = drills.find(drill => drill.id === training.drills[idx]);
+    navigation.navigate('DrillPageMinimal', { drill, training });
+  };
+
+  const goToFullDrill = () => {
+    navigation.navigate('DrillPage', { drill });
+  };
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      headerRight: () => {
+        return (
+          <TouchableOpacity
+            style={styles.nextButton}
+            onPress={() => {
+              if (currentDrill === totalDrills - 1) {
+                navigation.navigate('TrainingPage', { training });
+              } else {
+                const nextDrill = drills.find(drill => drill.id === training.drills[currentDrill + 1]);
+                navigation.navigate('DrillPageMinimal', { drill: nextDrill, training });
+              }
+            }}
+          >
+            <Image style={styles.nextButtonImg} source={iconCheckMark} />
+          </TouchableOpacity>
+        );
+      },
+    });
+  }, [drills, training, currentDrill, totalDrills, navigation]);
+
+  return (
+    <ScrollView style={styles.drillPage}>
+      <View style={styles.titleContainer}>
+        <Text style={styles.title}>
+          Drill {currentDrill + 1} - {drill.title}
+        </Text>
+        <Progress total={totalDrills} current={currentDrill + 1} onDotPress={onProgressDotPress} />
+      </View>
+      <View style={styles.description}>
```

Comment:
Reading the component, I felt like everything below this comment is the `MinimalDrill` (its responsability is to display attributes of the drill) and the training management could be a `LiveTrainingPage` that displays basic training info (title/progressbar) and manages a `currentDrill` that is passed as a props to `MinimalDrill`.

From the wireframe, it seems we'd like to add what I call `MinimalDrill` to our TrainingPage as well

Not sure if my comment is clear... What do you think?



<img width="694" alt="Screenshot 2020-04-24 at 21 23 10" src="https://user-images.githubusercontent.com/3810795/80249263-d15a3d80-8671-11ea-8f26-ad226fc162ec.png">

Reply:
Good idea. On it